api/envoy/extensions/filters/network/dubbo_proxy/v3/route.proto

```diff
@@ -26,7 +26,15 @@ message RouteConfiguration {
   // The name of the route configuration. Reserved for future use in asynchronous route discovery.
   string name = 1;
 
-  // The interface name of the service.
+  // The interface name of the service. Wildcard interface are supported in the suffix or prefix form.
+  // e.g. ``*.methods.add`` will match ``com.dev.methods.add``, ``com.prod.methods.add``, etc.
+  // ``com.dev.methods.*`` will match ``com.dev.methods.add``, ``com.dev.methods.update``, etc.
+  // Special wildcard ``*`` matching any interface.
+  //
+  // .. note::
+  //
+  //  The wildcard will not match the empty string.
+  //  e.g. ``*.methods.add`` will match ``com.dev.methods.add`` but not ``.methods.add``.
```

Comment:
Would there be any benefit to move to `StringMatcher` here? We have a bunch of discussions happening around things like wildcard matching on HTTP routes, see https://github.com/envoyproxy/envoy/issues/7763. This would allow us to intercept that work for free.

Reply:
@htuch The dubbo interface is often a full path to a JAVA type. In most cases, prefix- or suffix-based wildcard matching is sufficient.
Of course, if we use `StringMatcher`, then we can do more flexible matching operations. But it would require the introduction of a new configuration item.
If you suggest it, I can redesign the API to apply a new `StringMatcher` type field to match the dubbo interface.

Reply:
This was really just a suggestion for you folks to consider. I think its fine if you want to continue down the path you are on in this PR, since this is broadly consistent with how we do virtual hosts in regular HTTP.